tests.foreman.cli.test_host.HostCreateTestCase.test_positive_add_interface_by_id: Domain can't be blank

For more info see https://bugzilla.redhat.com/show_bug.cgi?id=1585734#c2 - looks like org/loc and domain's org/loc does not match.

```
Error Message

robottelo.cli.factory.CLIFactoryError: Failed to create Host with data: {   "architecture": null,   "architecture-id": "1",   "ask-root-password": null,   "autoheal": null,   "build": null,   "comment": null,   "compute-attributes": null,   "compute-profile": null,   "compute-profile-id": null,   "compute-resource": null,   "compute-resource-id": null,   "content-source-id": null,   "content-view": null,   "content-view-id": null,   "domain": null,   "domain-id": "94",   "enabled": null,   "environment": null,   "environment-id": null,   "hostgroup": null,   "hostgroup-id": null,   "hostgroup-title": null,   "hypervisor-guest-uuids": null,   "image": null,   "image-id": null,   "interface": null,   "ip": "192.56.82.236",   "kickstart-repository-id": null,   "lifecycle-environment": null,   "lifecycle-environment-id": null,   "location": null,   "location-id": "2",   "mac": "7e:41:e0:4d:a8:75",   "managed": null,   "medium": null,   "medium-id": "114",   "model": null,   "model-id": null,   "name": "wmxNjtUcJe",   "operatingsystem": null,   "operatingsystem-id": "1",   "organization": null,   "organization-id": "1",   "overwrite": null,   "owner": null,   "owner-id": null,   "owner-type": null,   "parameters": null,   "partition-table": null,   "partition-table-id": "94",   "progress-report-id": null,   "provision-method": null,   "puppet-ca-proxy": null,   "puppet-ca-proxy-id": null,   "puppet-class-ids": null,   "puppet-classes": null,   "puppet-proxy": null,   "puppet-proxy-id": null,   "pxe-loader": null,   "realm": null,   "realm-id": null,   "root-password": "SGxHGBRQ",   "service-level": null,   "subnet": null,   "subnet-id": null,   "volume": null } Command "host create" finished with return_code 65 stderr contains following message: [ERROR 2018-06-23T05:55:54 API] 422 Unprocessable Entity [ERROR 2018-06-23T05:55:54 Exception] Domain can't be blank Interfaces.domain can't find domain with this id Could not create the host:   Domain can't be blank   Interfaces.domain can't find domain with this id

Stacktrace

self = <tests.foreman.cli.test_host.HostCreateTestCase testMethod=test_positive_add_interface_by_id>

    @tier1
    def test_positive_add_interface_by_id(self):
        """New network interface can be added to existing host
    
            :id: e97dba92-61eb-47ad-a7d7-5f989292b12a
    
            :expectedresults: Interface added to host correctly and has proper
                domain and mac address
    
            :CaseImportance: Critical
            """
        domain = make_domain()
        mac = gen_mac(multicast=False)
        host = make_fake_host({
>           u'domain-id': domain['id'],
        })

tests/foreman/cli/test_host.py:209: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
robottelo/decorators/__init__.py:190: in cacheable_function
    new_object = func(options)
robottelo/cli/factory.py:1627: in make_fake_host
    return make_host(options)
robottelo/decorators/__init__.py:190: in cacheable_function
    new_object = func(options)
robottelo/cli/factory.py:1547: in make_host
    return create_object(Host, args, options)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cli_object = <class 'robottelo.cli.host.Host'>
options = {'architecture': None, 'architecture-id': '1', 'ask-root-password': None, 'autoheal': None, ...}
values = {'architecture-id': '1', 'domain-id': '94', 'location-id': '2', 'medium-id': '114', ...}

    def create_object(cli_object, options, values):
        """
        Creates <object> with dictionary of arguments.
    
        :param cli_object: A valid CLI object.
        :param dict options: The default options accepted by the cli_object
            create
        :param dict values: Custom values to override default ones.
        :raise robottelo.cli.factory.CLIFactoryError: Raise an exception if object
            cannot be created.
        :rtype: dict
        :return: A dictionary representing the newly created resource.
    
        """
        if values:
            diff = set(values.keys()).difference(set(options.keys()))
            if diff:
                logger.debug(
                    "Option(s) {0} not supported by CLI factory. Please check for "
                    "a typo or update default options".format(diff)
                )
        update_dictionary(options, values)
        try:
            result = cli_object.create(options)
        except CLIReturnCodeError as err:
            # If the object is not created, raise exception, stop the show.
            raise CLIFactoryError(
                u'Failed to create {0} with data:\n{1}\n{2}'.format(
                    cli_object.__name__,
                    json.dumps(options, indent=2, sort_keys=True),
>                   err.msg,
                )
            )
E           robottelo.cli.factory.CLIFactoryError: Failed to create Host with data:
E           {
E             "architecture": null,
E             "architecture-id": "1",
E             "ask-root-password": null,
E             "autoheal": null,
E             "build": null,
E             "comment": null,
E             "compute-attributes": null,
E             "compute-profile": null,
E             "compute-profile-id": null,
E             "compute-resource": null,
E             "compute-resource-id": null,
E             "content-source-id": null,
E             "content-view": null,
E             "content-view-id": null,
E             "domain": null,
E             "domain-id": "94",
E             "enabled": null,
E             "environment": null,
E             "environment-id": null,
E             "hostgroup": null,
E             "hostgroup-id": null,
E             "hostgroup-title": null,
E             "hypervisor-guest-uuids": null,
E             "image": null,
E             "image-id": null,
E             "interface": null,
E             "ip": "192.56.82.236",
E             "kickstart-repository-id": null,
E             "lifecycle-environment": null,
E             "lifecycle-environment-id": null,
E             "location": null,
E             "location-id": "2",
E             "mac": "7e:41:e0:4d:a8:75",
E             "managed": null,
E             "medium": null,
E             "medium-id": "114",
E             "model": null,
E             "model-id": null,
E             "name": "wmxNjtUcJe",
E             "operatingsystem": null,
E             "operatingsystem-id": "1",
E             "organization": null,
E             "organization-id": "1",
E             "overwrite": null,
E             "owner": null,
E             "owner-id": null,
E             "owner-type": null,
E             "parameters": null,
E             "partition-table": null,
E             "partition-table-id": "94",
E             "progress-report-id": null,
E             "provision-method": null,
E             "puppet-ca-proxy": null,
E             "puppet-ca-proxy-id": null,
E             "puppet-class-ids": null,
E             "puppet-classes": null,
E             "puppet-proxy": null,
E             "puppet-proxy-id": null,
E             "pxe-loader": null,
E             "realm": null,
E             "realm-id": null,
E             "root-password": "SGxHGBRQ",
E             "service-level": null,
E             "subnet": null,
E             "subnet-id": null,
E             "volume": null
E           }
E           Command "host create" finished with return_code 65
E           stderr contains following message:
E           [ERROR 2018-06-23T05:55:54 API] 422 Unprocessable Entity
E           [ERROR 2018-06-23T05:55:54 Exception] Domain can't be blank
E           Interfaces.domain can't find domain with this id
E           Could not create the host:
E             Domain can't be blank
E             Interfaces.domain can't find domain with this id

robottelo/cli/factory.py:140: CLIFactoryError

Standard Output

2018-06-23 05:55:27 - robottelo - DEBUG - Started Test: HostCreateTestCase/test_positive_add_interface_by_id
2018-06-23 05:55:27 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62162e80
2018-06-23 05:55:27 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:27 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv proxy list --search="url = https://<fqdn>:9090" --per-page="10000"'
2018-06-23 05:55:29 - robottelo.ssh - INFO - <<< stdout
Id,Name,URL,Features
1,<fqdn>,https://<fqdn>:9090,"Pulp, TFTP, DNS, DHCP, Puppet, Puppet CA, Logs, Dynflow, SSH, Discovery, Ansible, Openscap, Realm"

2018-06-23 05:55:29 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62162e80
2018-06-23 05:55:29 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62162dd8
2018-06-23 05:55:29 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:29 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv domain create --name="yibt71dw6v"'
2018-06-23 05:55:31 - robottelo.ssh - INFO - <<< stdout
Message,Id,Name
Domain [yibt71dw6v] created.,94,yibt71dw6v

2018-06-23 05:55:31 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62162dd8
2018-06-23 05:55:31 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b6202ce80
2018-06-23 05:55:31 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:31 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  domain info --id="94"'
2018-06-23 05:55:34 - robottelo.ssh - INFO - <<< stdout
Id:          94
Name:        yibt71dw6v
Description: 
DNS Id:      
Subnets:     

Parameters:  

Created at:  2018/06/23 09:55:31
Updated at:  2018/06/23 09:55:31


2018-06-23 05:55:34 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b6202ce80
2018-06-23 05:55:34 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62227198
2018-06-23 05:55:34 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:34 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  organization info --name="Default Organization"'
2018-06-23 05:55:37 - robottelo.ssh - INFO - <<< stdout
Id:                     1
Title:                  Default Organization
Name:                   Default Organization
Description:            
Users:                  

Smart proxies:          
    <fqdn>
Subnets:                

Compute resources:      

Installation media:     
    CentOS mirror
    CoreOS mirror
    Debian mirror
    Fedora Atomic mirror
    Fedora mirror
    FreeBSD mirror
    OpenSUSE mirror
    RancherOS mirror
    Ubuntu mirror
Templates:              
    Alterator default (provision)
    Alterator default finish (finish)
    Alterator default PXELinux (PXELinux)
    alterator_pkglist ()
    ansible_provisioning_callback ()
    ansible_tower_callback_script ()
    ansible_tower_callback_service ()
    Atomic Kickstart default (provision)
    AutoYaST default (provision)
    AutoYaST default iPXE (iPXE)
    AutoYaST default PXELinux (PXELinux)
    AutoYaST default user data (user_data)
    AutoYaST SLES default (provision)
    bmc_nic_setup ()
    Boot disk iPXE - generic host (Bootdisk)
    Boot disk iPXE - host (Bootdisk)
    chef_client ()
    coreos_cloudconfig ()
    CoreOS provision (provision)
    CoreOS PXELinux (PXELinux)
    create_users ()
    Discovery Debian kexec (kexec)
    Discovery Red Hat kexec (kexec)
    epel ()
    fix_hosts ()
    FreeBSD (mfsBSD) finish (finish)
    FreeBSD (mfsBSD) provision (provision)
    FreeBSD (mfsBSD) PXELinux (PXELinux)
    freeipa_register ()
    Grubby default (script)
    http_proxy ()
    idm_register ()
    Jumpstart default (provision)
    Jumpstart default finish (finish)
    Jumpstart default PXEGrub (PXEGrub)
    Junos default finish (finish)
    Junos default SLAX (provision)
    Junos default ZTP config (ZTP)
    Kickstart default finish (finish)
    Kickstart default iPXE (iPXE)
    Kickstart default PXEGrub (PXEGrub)
    Kickstart default PXEGrub2 (PXEGrub2)
    Kickstart default PXELinux (PXELinux)
    Kickstart default user data (user_data)
    kickstart_ifcfg_bonded_interface ()
    kickstart_ifcfg_bond_interface ()
    kickstart_ifcfg_generic_interface ()
    kickstart_ifcfg_get_identifier_names ()
    kickstart_networking_setup ()
    Kickstart oVirt-RHVH (provision)
    Kickstart oVirt-RHVH PXELinux (PXELinux)
    NX-OS default POAP setup (POAP)
    Preseed default (provision)
    Preseed default finish (finish)
    Preseed default iPXE (iPXE)
    Preseed default PXEGrub2 (PXEGrub2)
    Preseed default PXELinux (PXELinux)
    Preseed default user data (user_data)
    preseed_networking_setup ()
    puppet.conf ()
    puppetlabs_repo ()
    puppet_setup ()
    pxegrub2_chainload ()
    PXEGrub2 default local boot (PXEGrub2)
    pxegrub2_discovery ()
    PXEGrub2 global default (PXEGrub2)
    pxegrub_chainload ()
    PXEGrub default local boot (PXEGrub)
    pxegrub_discovery ()
    PXEGrub global default (PXEGrub)
    PXELinux chain iPXE (PXELinux)
    PXELinux chain iPXE UNDI (PXELinux)
    pxelinux_chainload ()
    PXELinux default local boot (PXELinux)
    PXELinux default memdisk (PXELinux)
    pxelinux_discovery ()
    PXELinux global default (PXELinux)
    rancheros_cloudconfig ()
    RancherOS provision (provision)
    RancherOS PXELinux (PXELinux)
    redhat_register ()
    remote_execution_ssh_keys ()
    saltstack_minion ()
    saltstack_setup ()
    Satellite Atomic Kickstart Default (provision)
    Satellite Kickstart Default (provision)
    Satellite Kickstart Default Finish (finish)
    Satellite Kickstart Default User Data (user_data)
    UserData default (user_data)
    WAIK default PXELinux (PXELinux)
    XenServer default answerfile (provision)
    XenServer default finish (finish)
    XenServer default PXELinux (PXELinux)
Domains:                

Environments:           
    common
    development
    production
Hostgroups:             

Parameters:             

Created at:             2018/06/23 07:54:32
Updated at:             2018/06/23 07:54:32
Label:                  Default_Organization
Description:            
Red Hat Repository URL: https://cdn.redhat.com
Service Levels:


2018-06-23 05:55:37 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62227198
2018-06-23 05:55:37 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62227198
2018-06-23 05:55:37 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:37 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  location info --name="Default Location"'
2018-06-23 05:55:40 - robottelo.ssh - INFO - <<< stdout
Id:                 2
Title:              Default Location
Name:               Default Location
Description:        
Users:              

Smart proxies:      
    <fqdn>
Subnets:            

Compute resources:  

Installation media: 
    92Dpbw
    CentOS mirror
    CoreOS mirror
    Debian mirror
    F3Jogw
    Fedora Atomic mirror
    Fedora mirror
    FreeBSD mirror
    mTe3Fk
    OpenSUSE mirror
    RancherOS mirror
    rQNNw8
    Ubuntu mirror
    yoVN0A
Templates:          
    Alterator default (provision)
    Alterator default finish (finish)
    Alterator default PXELinux (PXELinux)
    alterator_pkglist ()
    ansible_provisioning_callback ()
    ansible_tower_callback_script ()
    ansible_tower_callback_service ()
    Atomic Kickstart default (provision)
    AutoYaST default (provision)
    AutoYaST default iPXE (iPXE)
    AutoYaST default PXELinux (PXELinux)
    AutoYaST default user data (user_data)
    AutoYaST SLES default (provision)
    bmc_nic_setup ()
    Boot disk iPXE - generic host (Bootdisk)
    Boot disk iPXE - host (Bootdisk)
    chef_client ()
    coreos_cloudconfig ()
    CoreOS provision (provision)
    CoreOS PXELinux (PXELinux)
    create_users ()
    epel ()
    fix_hosts ()
    FreeBSD (mfsBSD) finish (finish)
    FreeBSD (mfsBSD) provision (provision)
    FreeBSD (mfsBSD) PXELinux (PXELinux)
    freeipa_register ()
    Grubby default (script)
    http_proxy ()
    Jumpstart default (provision)
    Jumpstart default finish (finish)
    Jumpstart default PXEGrub (PXEGrub)
    Junos default finish (finish)
    Junos default SLAX (provision)
    Junos default ZTP config (ZTP)
    Kickstart default finish (finish)
    Kickstart default iPXE (iPXE)
    Kickstart default PXEGrub (PXEGrub)
    Kickstart default PXEGrub2 (PXEGrub2)
    Kickstart default PXELinux (PXELinux)
    Kickstart default user data (user_data)
    kickstart_ifcfg_bonded_interface ()
    kickstart_ifcfg_bond_interface ()
    kickstart_ifcfg_generic_interface ()
    kickstart_ifcfg_get_identifier_names ()
    kickstart_networking_setup ()
    Kickstart oVirt-RHVH (provision)
    Kickstart oVirt-RHVH PXELinux (PXELinux)
    NX-OS default POAP setup (POAP)
    Preseed default (provision)
    Preseed default finish (finish)
    Preseed default iPXE (iPXE)
    Preseed default PXEGrub2 (PXEGrub2)
    Preseed default PXELinux (PXELinux)
    Preseed default user data (user_data)
    preseed_networking_setup ()
    puppet.conf ()
    puppetlabs_repo ()
    puppet_setup ()
    pxegrub2_chainload ()
    PXEGrub2 default local boot (PXEGrub2)
    pxegrub2_discovery ()
    PXEGrub2 global default (PXEGrub2)
    pxegrub_chainload ()
    PXEGrub default local boot (PXEGrub)
    pxegrub_discovery ()
    PXEGrub global default (PXEGrub)
    PXELinux chain iPXE (PXELinux)
    PXELinux chain iPXE UNDI (PXELinux)
    pxelinux_chainload ()
    PXELinux default local boot (PXELinux)
    PXELinux default memdisk (PXELinux)
    pxelinux_discovery ()
    PXELinux global default (PXELinux)
    rancheros_cloudconfig ()
    RancherOS provision (provision)
    RancherOS PXELinux (PXELinux)
    redhat_register ()
    remote_execution_ssh_keys ()
    saltstack_minion ()
    saltstack_setup ()
    UserData default (user_data)
    WAIK default PXELinux (PXELinux)
    XenServer default answerfile (provision)
    XenServer default finish (finish)
    XenServer default PXELinux (PXELinux)
Domains:            
    2wid5fvxwv
    bm55dvwta9
    igc5nxb21k
    in5wm5r8ml
    o12lcbbval
Environments:       
    common
    development
    KT_6LZOQo_Library_jEGrOAJjgV_25
    KT_BtVvtEwkYrbM_Library_klSDVDE_252
    KT_fye0OV_Library_EYrcxpcDWm_399
    KT_gARliDj_Library_PbHFji_140
    KT_GRIVIinaCh_Library_omHFBzBhjJG_237
    KT_hbZEbNmn_Library_KLWCMdwQ_4
    KT_hbZEbNmn_Library_LVOSEYY_3
    KT_l5HkO7_Library_dUKMImMZvx_395
    KT_LrSnKoIeJLT_Library_SqXVHHcCMg_163
    KT_NpnlzPXNJaU_Library_GbXsaOy_100
    KT_psuJTS_Library_XwGsPKxCcu_334
    KT_rzmRc2_Library_PxxzHiAyHT_23
    KT_U7ZDja_Library_cccrZWrvAv_17
    KT_U7ZDja_Library_XgWEejTucn_18
    KT_ufpEaQ_Library_cqnCxahDULfP_343
    KT_VvWWLWO_Library_gRZgONet_196
    KT_WdNIks_Library_mprTXwLCstd_117
    KT_y9R5CI_Library_ESxdAFJUfH_386
    KT_yHtFxJnBzbI_Library_RsapmOYjKO_247
    KT_ZhyiwbFJj_Library_amdmkF_127
    KT_ZMsvENJ_Library_oFqBEC_225
    production
Hostgroups:         

Parameters:         

Created at:         2018/06/23 07:54:32
Updated at:         2018/06/23 07:54:32


2018-06-23 05:55:40 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62227198
2018-06-23 05:55:40 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62227240
2018-06-23 05:55:40 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:40 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  architecture info --name="x86_64"'
2018-06-23 05:55:43 - robottelo.ssh - INFO - <<< stdout
Id:                1
Name:              x86_64
Operating systems: 
    RedHat 7.5
Created at:        2018/06/23 07:54:32
Updated at:        2018/06/23 07:54:32


2018-06-23 05:55:43 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62227240
2018-06-23 05:55:43 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62628ac8
2018-06-23 05:55:43 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:43 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv os list --search="name="RedHat" AND major="6" OR major="7"" --per-page="10000"'
2018-06-23 05:55:45 - robottelo.ssh - INFO - <<< stdout
Id,Title,Release name,Family
1,RedHat 7.5,,Redhat

2018-06-23 05:55:45 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62628ac8
2018-06-23 05:55:45 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62227240
2018-06-23 05:55:45 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:45 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv partition-table list --operatingsystem-id="1" --per-page="10000"'
2018-06-23 05:55:47 - robottelo.ssh - INFO - <<< stdout
Id,Name,OS Family
94,Kickstart default,Redhat

2018-06-23 05:55:47 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62227240
2018-06-23 05:55:48 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62147c50
2018-06-23 05:55:48 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:48 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv medium create --location-ids="2" --name="0KdQD0" --operatingsystem-ids="1" --organization-ids="1" --path="http://jrmekp"'
2018-06-23 05:55:50 - robottelo.ssh - INFO - <<< stdout
Message,Id,Name
Installation medium created.,114,0KdQD0

2018-06-23 05:55:50 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62147c50
2018-06-23 05:55:50 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62153a20
2018-06-23 05:55:50 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:50 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  medium info --id="114"'
2018-06-23 05:55:52 - robottelo.ssh - INFO - <<< stdout
Id:                114
Name:              0KdQD0
Path:              http://jrmekp
OS Family:         
Operating systems: 
    RedHat 7.5
Locations:         
    Default Location
Organizations:     
    Default Organization
Created at:        2018/06/23 09:55:49
Updated at:        2018/06/23 09:55:49


2018-06-23 05:55:52 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62153a20
2018-06-23 05:55:52 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f5b62159e80
2018-06-23 05:55:52 - robottelo.ssh - INFO - Connected to [<fqdn>]
2018-06-23 05:55:52 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv host create --architecture-id="1" --domain-id="94" --ip="192.56.82.236" --location-id="2" --mac="7e:41:e0:4d:a8:75" --medium-id="114" --name="wmxNjtUcJe" --operatingsystem-id="1" --organization-id="1" --partition-table-id="94" --root-password="SGxHGBRQ"'
2018-06-23 05:55:55 - robottelo.ssh - INFO - <<< stderr
[ERROR 2018-06-23T05:55:54 API] 422 Unprocessable Entity
[ERROR 2018-06-23T05:55:54 Exception] Domain can't be blank
Interfaces.domain can't find domain with this id
Could not create the host:
  Domain can't be blank
  Interfaces.domain can't find domain with this id

2018-06-23 05:55:55 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f5b62159e80
2018-06-23 05:55:55 - robottelo - DEBUG - Finished Test: HostCreateTestCase/test_positive_add_interface_by_id
```

--- a/tests/foreman/cli/test_host.py
+++ b/tests/foreman/cli/test_host.py
@@ -210,7 +210,10 @@ class HostCreateTestCase(CLITestCase):
 
         :CaseImportance: Critical
         """
-        domain = make_domain()
+        domain = make_domain({
+            u'organizations': u'Default Organization',
+            u'locations': u'Default Location'
+        })
         mac = gen_mac(multicast=False)
         host = make_fake_host({
             u'domain-id': domain['id'],
@@ -219,6 +222,7 @@ class HostCreateTestCase(CLITestCase):
             u'host-id': host['id'],
             u'domain-id': domain['id'],
             u'mac': mac,
+            u'type': u'interface'
         })
         host = Host.info({u'id': host['id']})
         host_interface = HostInterface.info({

